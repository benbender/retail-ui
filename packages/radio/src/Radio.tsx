import clsx from 'clsx'
import * as React from 'react'

import { RadioStyles } from './styles'

type ReactInputProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'disabled'
>
type Ref = HTMLInputElement

export type RadioProps = {
  isDisabled?: boolean
}

export const Radio = React.forwardRef<Ref, ReactInputProps & RadioProps>(
  (props, ref) => {
    const { children, className, isDisabled, ...rest } = props

    const cls = clsx(
      className,
      RadioStyles.base,
      isDisabled && RadioStyles.disabled,
    )
    const labelCls = clsx(RadioStyles.label)

    return (
      <div>
        <label className={labelCls}>
          <input
            className={cls}
            ref={ref}
            {...rest}
            type="radio"
            disabled={isDisabled}
          />
          <span
            className={`ml-2 ${isDisabled && `opacity-50 cursor-not-allowed`}`}
          >
            {children}
          </span>
        </label>
      </div>
    )
  },
)

Radio.displayName = 'Radio'
