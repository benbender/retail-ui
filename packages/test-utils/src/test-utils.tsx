import { defaultTheme, ThemeProvider } from '@retail-ui/theme'
import {
  render as RtlRender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react'
import * as React from 'react'

type UI = Parameters<typeof RtlRender>[0]

// UI-less passthrough fallback to prevent using conditional logic in render
function ChildrenPassthrough({ children }: { children: React.ReactElement }) {
  return children
}

export interface TestOptions extends Omit<RenderOptions, 'wrapper'> {
  /**
   * optional additional wrapper, e.g. Context
   *
   * @example
   * ```ts
   * // single wrapper
   * render(<MyConponent />, {
   *  wrapper: MyContext
   * });
   *
   * // multiple wrapper
   * render(<MyConponent />, {
   *  wrapper: ({ children }) => (
   *    <ContextA>
   *      <ContextB>
   *        {children}
   *      <ContextB />
   *    <ContextA />
   *  )
   * });
   *
   * ```
   */
  wrapper?: typeof ChildrenPassthrough
}

/**
 * Custom render for @testing-library/react
 *
 * @see https://testing-library.com/docs/react-testing-library/setup#custom-render
 * @param component the component under test
 * @param options customized test options
 */
export const render = (
  ui: UI,
  { wrapper: Wrapper = ChildrenPassthrough, ...options }: TestOptions = {},
): RenderResult => {
  return RtlRender(
    <ThemeProvider value={{ theme: defaultTheme }}>
      <Wrapper>{ui}</Wrapper>
    </ThemeProvider>,
    options,
  )
}

export { fireEvent } from '@testing-library/react'
