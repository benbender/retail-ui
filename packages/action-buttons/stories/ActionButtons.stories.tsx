import * as React from 'react'

import { ActionButtons, ActionButtonsProps } from '../src'

const meta = {
  title: 'Retail-UI/ActionButtons',
  component: ActionButtons,
}

export default meta

const Template = (args: ActionButtonsProps) => (
  <ActionButtons {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {}
