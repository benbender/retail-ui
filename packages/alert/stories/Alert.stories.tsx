import * as React from 'react'

import { Alert, AlertProps } from '../src'

const meta = {
  title: 'Retail-UI/Alert',
  component: Alert,
}

export default meta

const Template = (args: AlertProps) => (
  <Alert {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  children: 'A alert that informs you of stuff',
}
