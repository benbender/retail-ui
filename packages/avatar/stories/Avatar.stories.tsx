import * as React from 'react'

import { Avatar, AvatarProps } from '../src'

const meta = {
  title: 'Retail-UI/Avatar',
  component: Avatar,
}

export default meta

const Template = (args: AvatarProps) => (
  <Avatar {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  src: `https://pbs.twimg.com/profile_images/1285102166709919744/XvlWrJny_400x400.jpg`,
}
