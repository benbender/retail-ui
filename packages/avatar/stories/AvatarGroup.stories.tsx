import * as React from 'react'

import { Avatar, AvatarGroup, AvatarProps } from '../src'

const meta = {
  title: 'Retail-UI/AvatarGroup',
  component: AvatarGroup,
}

export default meta

const src = `https://pbs.twimg.com/profile_images/1285102166709919744/XvlWrJny_400x400.jpg`

const Group = (args: AvatarProps) => (
  <AvatarGroup data-testid={meta.title}>
    <Avatar src={src} {...args} />
    <Avatar title="PJ" size="base" {...args} />
    {/* <Avatar icon={UserSolid} {...args} /> */}
  </AvatarGroup>
)

export const Simple = Group.bind({})
// @ts-ignore
Simple.args = {}
