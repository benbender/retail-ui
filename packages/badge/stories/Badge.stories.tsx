import * as React from 'react'

import { Badge, BadgeProps } from '../src'

const meta = {
  title: 'Retail-UI/Badge',
  component: Badge,
}

export default meta

const Template = (args: BadgeProps) => (
  <Badge {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  children: 'Badge',
}
