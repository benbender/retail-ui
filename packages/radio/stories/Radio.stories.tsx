import * as React from 'react'

import { Radio, RadioProps } from '../src'

const meta = {
  title: 'Retail-UI/Radio',
  component: Radio,
}

export default meta

const Template = (args: RadioProps) => (
  <Radio {...args} data-testid={meta.title} value="busines" />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  children: 'Radio',
}
