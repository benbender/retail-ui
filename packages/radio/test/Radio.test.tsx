import { render } from '@retail-ui/test-utils'
import * as React from 'react'

import * as Radio from '../stories/Radio.stories'

describe('Radio', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Radio.Simple />)
    expect(asFragment()).toMatchSnapshot()
  })
})
