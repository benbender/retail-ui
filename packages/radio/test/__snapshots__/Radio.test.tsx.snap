// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<DocumentFragment>
  <div>
    <label
      class="inline-flex items-center text-gray-600 dark:text-gray-400"
    >
      <input
        class="text-purple-600 form-radio focus:border-purple-400 focus:outline-none dark:border-gray-600 dark:bg-gray-700 focus:shadow-outline-purple dark:focus:border-gray-600 dark:focus:shadow-outline-gray"
        data-testid="Retail-UI/Radio"
        type="radio"
        value="busines"
      />
      <span
        class="ml-2 undefined"
      />
    </label>
  </div>
</DocumentFragment>
`;
